File: src/net/auoeke/cin/lexer/lexeme/WhitespaceLexeme.java
package net.auoeke.cin.lexer.lexeme;

import net.auoeke.cin.lexer.error.SyntaxError;

public final class WhitespaceLexeme extends Lexeme {
    public final String value;

    public WhitespaceLexeme(int line, int column, String value, SyntaxError error) {
        super(line, column, error);

        this.value = value;
    }

    public WhitespaceLexeme(int line, int column, String value) {
        this(line, column, value, null);
    }

    @Override public Type type() {
        return Type.WHITESPACE;
    }

    @Override public String toString() {
        return this.value;
    }
}

File: src/net/auoeke/cin/lexer/lexeme/WhitespaceLexeme.kt
package net.auoeke.cin.lexer.lexeme

import net.auoeke.extensions.string

class WhitespaceLexeme(line: Int, column: Int, val value: String) : Lexeme(line, column) {
    override val type: Type get() = Type.WHITESPACE

    override fun toString(): String = value.string
}
